packages/blockprotocol/cli/commands/publish/find-api-key.js

```diff
@@ -0,0 +1,99 @@
+import chalk from "chalk";
+import fs from "fs-extra";
+import path from "node:path";
+
+import { printSpacer } from "../../print-spacer.js";
+import { doesUserAgree } from "./does-user-agree.js";
+import { findProjectRoot } from "./shared/find-project-root.js";
+
+const environmentVariableName = "BLOCK_PROTOCOL_API_KEY";
+const configFileName = ".bprc.json";
+const configFileKey = "apiKey";
+const configFileTemplate = {
+  [configFileKey]:
+    "b10ck5.00000000000000000000000000000000.00000000-0000-0000-0000-000000000000",
+};
+
+/**
+ * Retrieve an API key from a provided JSON file
+ * @param {string} filePath
+ */
+const extractKeyFromJsonFile = (filePath) => {
+  let parsedJson;
+  try {
+    parsedJson = JSON.parse(fs.readFileSync(filePath));
+  } catch (err) {
+    console.log(`Could not parse config file: ${chalk.red(err.message)}`);
+    console.log(`Path: ${filePath}`);
+    process.exit();
+  }
+  const key = parsedJson[configFileKey];
+
+  if (!key) {
+    console.log(
+      `Config file ${chalk.red("does not contain")} '${configFileKey}' key.`,
+    );
+    console.log(`Path: ${filePath}`);
+    process.exit();
+  }
+
+  console.log(chalk.green(`Found API key in configuration file.`));
+  return key;
+};
+
+export const findApiKey = async () => {
+  const keyInEnvironment = process.env[environmentVariableName];
+
+  if (keyInEnvironment) {
+    console.log(chalk.green("Found API key in environment."));
+    return keyInEnvironment;
+  }
+
+  // see if there's a config file in the current working directory
+  const configFilePathInCwd = path.resolve("./", configFileName);
+  const fileExistsInCwd = await fs.pathExists(configFilePathInCwd);
+  if (fileExistsInCwd) {
+    return extractKeyFromJsonFile(configFilePathInCwd);
+  }
+
+  // if not, find the project root and check for the file there
+  const projectRootPath = await findProjectRoot();
```

Comment:
Not sure we need to search for project root in this case. AFAIU, config files are searched upwards till `/` is reached. For example, `~/.bprc.json` can be a valid location for the config.

Reply:
Fair enough – simplified in https://github.com/blockprotocol/blockprotocol/pull/575/commits/c9d54518a8f1b58b023e7d4c72df8529552fbe96